### PR TITLE
Feat/list asset logs 163048333

### PR DESCRIPTION
**Pull request by andela-fojuri**
## What does this PR do?
- Displays all asset logs grouped by date

## Description of Task to be completed?
Given an admin with access,
When logs page is clicked,
Then I should see check-in/checkout logs for all assets grouped by date.

## How should this be manually tested?
- run the app and click on `Logs` on the Navbar

## What are the relevant pivotal tracker stories?
[#163048333](https://www.pivotaltracker.com/story/show/163048333)

## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [x] Test
- [ ] Documentation
## Screenshots
screen shot 2019-03-07 at 1 06 11 pm

### CODE REVIEW

File: src/__test__/actions/assetLogs.actions.test.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import constants from '../../_constants';
+
+import loadAssetLogs from '../../_actions/assetLogs.actions';
+
+import assetLogs from '../../_mock/assetLogs';
+
+const {
+  LOAD_ASSET_LOGS_SUCCESS,
+  LOAD_ASSET_LOGS_FAILURE,
+  LOADING_ASSETS_LOGS
```
Comment (skboadu) [inline src/__test__/actions/assetLogs.actions.test.js]: Might help to make the naming consistent here - `LOADING_ASSET_LOGS`

File: src/__test__/actions/assetLogs.actions.test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  beforeEach(() => {
+    mock = new MockAdapter(axios);
+    store = mockStore({});
+  });
+
+  afterEach(() => {
+    store.clearActions();
+  });
+
+  it('should dispatch LOADING_ASSET_LOGS with isLoading false when done fetching asset logs', () => {
```
Comment (skboadu) [inline src/__test__/actions/assetLogs.actions.test.js]: For jest async tests, you should consider using `expect.hasAssertions` or similar to ensure it fails if assertions don't fire

File: src/__test__/components/AssetLogComponent.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import assetLogs from '../../_mock/assetMakes';
+
+describe('Renders <AssetLogComponent /> correctly', () => {
+  const props = {
+    loadAssetLogs: jest.fn(),
+    isLoading: false,
+    assetLogs
+  };
+  let wrapper = shallow(<AssetLogComponent
```
Comment (skboadu) [inline src/__test__/components/AssetLogComponent.test.js]: This should be on a single line

File: src/__test__/reducers/assetLogs.reducer.test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// third-party library
```
Comment (skboadu) [inline src/__test__/reducers/assetLogs.reducer.test.js]: Not sure what this comment is for but the code is self-explanatory so comment not needed

File: src/__test__/reducers/assetLogs.reducer.test.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import expect from 'expect';
+
+// reducer
```
Comment (skboadu) [inline src/__test__/reducers/assetLogs.reducer.test.js]: See above

File: src/__test__/reducers/assetLogs.reducer.test.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import assetLogReducer from '../../_reducers/assetLogs.reducer';
+
+// initial mock State
```
Comment (skboadu) [inline src/__test__/reducers/assetLogs.reducer.test.js]: See above

File: src/__test__/reducers/assetLogs.reducer.test.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import mockStore from '../../_mock/mockStore';
+
+// mock data
```
Comment (skboadu) [inline src/__test__/reducers/assetLogs.reducer.test.js]: See above

File: src/__test__/actions/assetLogs.actions.test.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+import assetLogs from '../../_mock/assetLogs';
+
+const {
+  LOAD_ASSET_LOGS_SUCCESS,
+  LOAD_ASSET_LOGS_FAILURE,
+  LOADING_ASSETS_LOGS
+} = constants;
+
+const middleware = [thunk];
+const mockStore = configureMockStore(middleware);
```
Comment (skboadu) [inline src/__test__/actions/assetLogs.actions.test.js]: Why aren't you importing and using the mockStore used in the reducer tests?

File: src/components/AssetLogs/AssetLogComponent.jsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+  async componentDidMount() {
+    await this.props.loadAssetLogs();
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({
+      groupedAssetLogs:
+      _.groupBy(this.props.assetLogs, asset => new Date(asset.created_at).toDateString())
+    });
+  }
+
+  con = index => (
```
Comment (skboadu) [inline src/components/AssetLogs/AssetLogComponent.jsx]: What does `con` refer to? I prefer readable variable names

File: src/__test__/actions/assetLogs.actions.test.js (hunk 8 above)
Comment (andela-fojuri) [reply src/__test__/actions/assetLogs.actions.test.js]: Using redux thunk middleware based on  [this](https://redux.js.org/recipes/writing-tests#async-action-creators) documentation.

**Pull request merged**